ApplicationLibCode/Commands/PlotBuilderCommands/RicAppendSummaryPlotsForObjectsFeature.cpp

```diff
@@ -0,0 +1,292 @@
+                if ( !curveSets.empty() )
+                {
+                    for ( auto curveSet : curveSets )
+                    {
+                        auto sourceAddress = curveSet->summaryAddress();
+                        auto modifiedAdr   = modifyAddress( sourceAddress, summaryAdrCollection );
+
+                        curveSet->setSummaryAddress( modifiedAdr );
+                    }
+                }
+                else
+                {
+                    auto curves = duplicatedPlot->allCurves( RimSummaryDataSourceStepping::Axis::Y_AXIS );
+                    for ( auto c : curves )
+                    {
+                        auto sourceAddress = c->summaryAddressY();
+                        auto modifiedAdr   = modifyAddress( sourceAddress, summaryAdrCollection );
+
+                        c->setSummaryAddressY( modifiedAdr );
+                    }
+                }
+
+                activePlotWindow->addPlot( duplicatedPlot );
+
+                duplicatedPlot->resolveReferencesRecursively();
+            }
+        }
+
+        activePlotWindow->loadDataAndUpdate();
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RicAppendSummaryPlotsForObjectsFeature::setupActionLook( QAction* actionToSetup )
+{
+    QString objectType = "Objects";
+
+    auto addresses = selectedCollections();
+
+    if ( !addresses.empty() )
+    {
+        auto firstAdr = addresses.front();
+        if ( firstAdr->contentType() == RimSummaryAddressCollection::CollectionContentType::WELL )
+        {
+            objectType = "Wells";
+        }
+        else if ( firstAdr->contentType() == RimSummaryAddressCollection::CollectionContentType::WELL_GROUP )
+        {
+            objectType = "Well Groups";
+        }
+        else if ( firstAdr->contentType() == RimSummaryAddressCollection::CollectionContentType::REGION )
+        {
+            objectType = "Regions";
+        }
+    }
+
+    auto text = QString( "Append Plots For " ) + objectType;
+    actionToSetup->setText( text );
+    actionToSetup->setIcon( QIcon( ":/SummaryPlotLight16x16.png" ) );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+std::vector<RimSummaryAddressCollection*> RicAppendSummaryPlotsForObjectsFeature::selectedCollections()
+{
+    std::vector<RimSummaryAddressCollection*> sumAddressCollections;
+    caf::SelectionManager::instance()->objectsByType( &sumAddressCollections );
+
+    if ( sumAddressCollections.size() == 1 )
+    {
+        auto coll = sumAddressCollections[0];
+        if ( coll->name() == "Wells" || coll->name() == "Regions" || coll->name() == "Well Groups" )
```

Comment:
Maybe extend CollectionContentType to include WELLS_FOLDER, REGIONS_FOLDER and WELLGROUPS_FOLDER to avoid name checks?